norse/task/cartpole.py

```diff
@@ -75,22 +74,15 @@ def __init__(self, device="cpu"):
 
     def forward(self, x):
         scale = 50
-        x = x.to(self.device)
         x_pos = self.constant_current_encoder(torch.nn.functional.relu(scale * x))
         x_neg = self.constant_current_encoder(torch.nn.functional.relu(-scale * x))
         x = torch.cat([x_pos, x_neg], dim=2)
 
         seq_length, batch_size, _ = x.shape
 
-        # state for hidden layer
-        s1 = self.lif.initial_state(batch_size, device=self.device)
-        # state for output layer
-        so = self.readout.initial_state(batch_size, device=self.device)
-
-        voltages = torch.zeros(
-            seq_length, batch_size, self.output_features, device=self.device
-        )
+        voltages = torch.zeros(seq_length, batch_size, self.output_features)
```

Comment:
I think this still needs a device parameter, maybe use self.parameters[0].device.

Reply:
I solved it with `x.device`